src/io/inputs/yml-system/converter.cpp

```diff
@@ -121,17 +122,106 @@ static SystemModel::Component createComponent(const YmlSystem::Component& c,
     return component;
 }
 
+/**
+ * @brief Creates a SystemModel::Connection from a YmlSystem::Connection and a map of components.
+ *
+ * This function constructs a SystemModel::Connection by looking up components and ports
+ * based on the IDs provided in the YmlSystem::Connection. It ensures that the ports are
+ * of the same type and that fields are correctly configured for sending and receiving.
+ *
+ * @param connection A YmlSystem::Connection object containing the connection details.
+ * @param components An unordered map of component IDs to SystemModel::Component objects.
+ *
+ * @return A SystemModel::Connection object representing the created connection.
+ *
+ * @throw std::invalid_argument if a component or port is not found, if the ports are not
+ *        of the same type, or if fields are incorrectly configured for sending/receiving.
+ */
+static SystemModel::Connection createConnection(
+  const YmlSystem::Connection& connection,
+  const std::unordered_map<std::string, SystemModel::Component>& components)
+{
+    auto findComponent = [&components](const std::string& id) -> const SystemModel::Component&
+    {
+        const auto it = components.find(id);
+        if (it == components.end())
+        {
+            throw std::invalid_argument("Component with id '" + id + "' not found in system.");
+        }
+        return it->second;
+    };
```

Comment:
Maybe we could have : 
```c++
auto findComponent = [&components](const std::string& id) -> const SystemModel::Component&
{
    if (! components.contains(id))
    {
        throw std::invalid_argument("Component with id '" + id + "' not found in system.");
    }
     return component[id];
};
```

Reply:
sorry, my code is more efficient 😆

Reply:
@a-zakir is right, the current implementation only requires one lookup

Reply:
True